subprojects/gradle/room-config/src/main/kotlin/com/avito/android/plugin/RoomConfigPlugin.kt

```diff
@@ -26,15 +31,29 @@ class RoomConfigPlugin : Plugin<Project> {
             target.dependencies.add("testImplementation", "androidx.room:room-testing:$archPersistence")
         }
 
-        target.plugins.withId("kotlin-kapt") {
+        target.plugins.withId("kotlin-kapt") { plugin ->
 
             target.dependencies.add("kapt", "androidx.room:room-compiler:$archPersistence")
 
             target.extensions.getByType<KaptExtension>().run {
                 arguments {
-                    arg("room.schemaLocation", File(target.projectDir, "room-schemas").absolutePath)
+                    arg("room.schemaLocation", File(target.projectDir, "room-schemas-$kaptIncremental").absolutePath)
+                    arg("room.incremental", kaptIncremental)
                 }
             }
+
+            if (kaptIncremental) {
+                target.afterEvaluate { checkIncrementalKaptSupport(target) }
+            }
         }
     }
+
+    private fun checkIncrementalKaptSupport(project: Project) {
+        val mode = project.getOptionalStringProperty(
+            name = "avito.room.incrementalKaptSupportCheck",
+            nullIfBlank = true
+        ) ?: "warning"
+        RoomIncrementalKaptSupportCheckMode.valueOf(mode.toUpperCase(Locale.ENGLISH)).check(project)
```

Comment:
How long does it take? We do it in a configuration phase

Reply:
I think I can measure it.

Reply:
Agree, we could only construct a RoomIncrementalKaptSupportCheckMode instance.
But wrap check to the task